core-api/src/main/java/com/optimizely/ab/notification/DecisionNotification.java

```diff
@@ -0,0 +1,167 @@
+
+        private String featureKey;
+        private Boolean featureEnabled;
+        private FeatureDecision featureDecision;
+        private String variableKey;
+        private FeatureVariable.VariableType variableType;
+        private Object variableValue;
+
+        protected FeatureVariableDecisionNotificationBuilder() {
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withUserId(String userId) {
+            this.userId = userId;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withAttributes(Map<String, ?> attributes) {
+            this.attributes = attributes;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withFeatureKey(String featureKey) {
+            this.featureKey = featureKey;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withFeatureEnabled(Boolean featureEnabled) {
+            this.featureEnabled = featureEnabled;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withFeatureDecision(FeatureDecision featureDecision) {
+            this.featureDecision = featureDecision;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withVariableKey(String variableKey) {
+            this.variableKey = variableKey;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withVariableType(FeatureVariable.VariableType variableType) {
+            this.variableType = variableType;
+            return this;
+        }
+
+        public FeatureVariableDecisionNotificationBuilder withVariableValue(Object variableValue) {
+            this.variableValue = variableValue;
+            return this;
+        }
+
+        public DecisionNotification build() {
+            decisionInfo = new HashMap<>();
+            decisionInfo.put(FEATURE_KEY, featureKey);
+            decisionInfo.put(FEATURE_ENABLED, featureEnabled);
+            decisionInfo.put(VARIABLE_KEY, variableKey);
+            decisionInfo.put(VARIABLE_TYPE, variableType);
+            decisionInfo.put(VARIABLE_VALUE, variableValue);
+            if (featureDecision.decisionSource != null && featureDecision.decisionSource.equals(FeatureDecision.DecisionSource.EXPERIMENT)) {
+                decisionInfo.put(SOURCE_EXPERIMENT_KEY, featureDecision.experiment.getKey());
+                decisionInfo.put(SOURCE_VARIATION_KEY, featureDecision.variation.getKey());
+                decisionInfo.put(SOURCE, featureDecision.decisionSource);
+            } else {
+                decisionInfo.put(SOURCE_EXPERIMENT_KEY, null);
+                decisionInfo.put(SOURCE_VARIATION_KEY, null);
+                decisionInfo.put(SOURCE, FeatureDecision.DecisionSource.ROLLOUT);
+            }
+
+            this.type = NotificationCenter.DecisionNotificationType.FEATURE_VARIABLE.toString();
```

Comment:
If `type` is not something that's settable, it's more clear if it's set in the constructor. Maybe it should even be parameter of `BaseDecisionNotificationBuilder` constructor.